intelmq/bots/collectors/tcp/collector.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+
+import socket
+
+import intelmq.lib.utils as utils
+from intelmq.lib.bot import CollectorBot
+
+
+class TCPCollectorBot(CollectorBot):
+
+    def init(self):
+        self.address = (self.parameters.ip, int(self.parameters.port))
+        self.separator = utils.encode(self.parameters.separator)
+        self.BUFFER_SIZE = 1024
+        self.connect()
+
+    def process(self):
+        conn = None
+        try:
+            conn, addr = self.con.accept()
+            print('Connection address:', addr)
+            data = b""
+            while True:
+                b = conn.recv(self.BUFFER_SIZE)
+                if not b:
+                    break
+
+                data += b
+                messages = data.split(self.separator)
+                if len(messages) > 1:
+                    self.logger.debug('Received parsable data: %s.', data)
+                    for event in messages:
+                        if event:
+                            report = self.new_report()
+                            report.add("raw", event)
+                            self.send_message(report)
+                    data = messages[-1]  # return back the non-finished message
+            conn.send(b"OK")
+        except socket.error as e:
+            self.logger.exception("Reconnecting.")
+            self.con.close()
+            self.connect()
+        except AttributeError as e:
+            self.logger.info('Attribute error.')
+        finally:
+            if conn:
+                conn.close()
+
+    def connect(self):
+        self.con = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.con.bind(self.address)
+        self.con.listen()
```

Comment:
I get this error when running the bot (also in the tests):
```
intelmqctl run tcp-collector 
intelmqctl: Starting tcp-collector...
tcp-collector: TCPCollectorBot initialized with id tcp-collector and intelmq 1.1.0.alpha1 and python 3.4.6 (default, Mar 22 2017, 12:26:13) [GCC] as process 27674.
tcp-collector: Bot is starting.
tcp-collector: Bot initialization failed.
Traceback (most recent call last):
  File "/home/sebastian/dev/intelmq/intelmq/lib/bot.py", line 82, in __init__
    self.init()
  File "/home/sebastian/dev/intelmq/intelmq/bots/collectors/tcp/collector.py", line 15, in init
    self.connect()
  File "/home/sebastian/dev/intelmq/intelmq/bots/collectors/tcp/collector.py", line 52, in connect
    self.con.listen()
TypeError: listen() takes exactly one argument (0 given)
tcp-collector: Bot stopped.
Bot exited with code 1.
```

Reply:
(I've corrected the executability.)

It seems you have Python 3.4-. Since 3.5, the argument is chosen automatically. I've changed the code to support Python 3.4.